pkgs/development/python-modules/bitsandbytes/default.nix

```diff
@@ -48,47 +52,51 @@ buildPythonPackage {
   inherit pname version;
   pyproject = true;
 
-  disabled = pythonOlder "3.7";
-
   src = fetchFromGitHub {
     owner = "TimDettmers";
     repo = "bitsandbytes";
     rev = "refs/tags/${version}";
-    hash = "sha256-GFbFKPdV96DXPA+PZO4h0zdBclN670fb0PGv4QPHWHU=";
+    hash = "sha256-yvxD5ymMK5p4Xg7Csx/90mPV3yxUC6QUuF/8BKO2p0k=";
   };
 
-  postPatch =
-    ''
-      substituteInPlace Makefile --replace "/usr/bin/g++" "g++" --replace "lib64" "lib"
-      substituteInPlace bitsandbytes/cuda_setup/main.py  \
-        --replace "binary_path = package_dir / self.binary_name"  \
-                  "binary_path = Path('$out/${python.sitePackages}/${pname}')/self.binary_name"
-    ''
-    + lib.optionalString torch.cudaSupport ''
-      substituteInPlace bitsandbytes/cuda_setup/main.py  \
-        --replace "/usr/local/cuda/lib64" "${cuda-native-redist}/lib"
-    '';
+  # By default, which library is loaded depends on the result of `torch.cuda.is_available()`.
+  # When `cudaSupport` is enabled, bypass this check and load the cuda library unconditionnally.
+  # Indeed, in this case, only `libbitsandbytes_cuda124.so` is built. `libbitsandbytes_cpu.so` is not.
+  # Also, hardcode the path to the previously built library instead of relying on
```

Comment:
Wow this really sounds like a bug in bitsanbytes, I'm surprised they're not reacting to the issue

Reply:
FWIW on our side we could actually make their presumably-inteded behaviour work, only at double the cost: we could just make the cuda-version depend on the cpu-version.

The question is, is the `xxxxx_cuXXX.so` library usable in absence of GPUs?

Reply:
I think that this current approach is fine. If you set `cudaSupport = true`, it means that you want to use cuda.



> The question is, is the `xxxxx_cuXXX.so` library usable in absence of GPUs?

I don't think so.

Reply:
> I think that this current approach is fine. If you set cudaSupport = true, it means that you want to use cuda.

Well the upstream has implemented this (broken?) dynamic dispatching logic so I guess they intended it the other way. Anyway, implementing that logic from scratch is not in-scope here

Reply:
> Well the upstream has implemented this (broken?) dynamic dispatching logic so I guess they intended it the other way. Anyway, implementing that logic from scratch is not in-scope here

Yes, indeed.